static/app/components/devtoolbar/components/replay/replayPanel.tsx

```diff
@@ -0,0 +1,126 @@
+import {Fragment, useEffect, useState} from 'react';
+import {css} from '@emotion/react';
+
+import {Button} from 'sentry/components/button';
+import SentryAppLink from 'sentry/components/devtoolbar/components/sentryAppLink';
+import {resetFlexRowCss} from 'sentry/components/devtoolbar/styles/reset';
+import ProjectBadge from 'sentry/components/idBadge/projectBadge';
+import {IconPause, IconPlay} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import type {PlatformKey} from 'sentry/types/project';
+
+import useConfiguration from '../../hooks/useConfiguration';
+import {panelInsetContentCss, panelSectionCss} from '../../styles/panel';
+import {smallCss} from '../../styles/typography';
+import PanelLayout from '../panelLayout';
+
+const TRUNC_ID_LENGTH = 16;
+const POLL_INTERVAL_MS = 3000;
+
+export default function ReplayPanel() {
+  const {projectSlug, projectId, projectPlatform, trackAnalytics, SentrySDK} =
+    useConfiguration();
+  const replay =
+    SentrySDK && 'getReplay' in SentrySDK ? SentrySDK.getReplay() : undefined;
+
+  const [replayId, setReplayId] = useState<string | undefined>(() =>
+    replay?.getReplayId()
+  );
```

Comment:
```suggestion
  const [replayId, setReplayId] = useState<string | undefined>(
    replay?.getReplayId()
  );
```
does this work?

Reply:
These are equivalent, but passing in a callback causes `getReplayId` to be called on less re-renders. https://medium.com/@jscodelover/using-functions-as-the-initial-state-in-the-usestate-hook-in-react-7069cdfcc6ed

I want this to refresh every time the panel is _first_ rendered (e.g. switching panels). Any thoughts @ryan953 ?

Reply:
Two things to disambiguate: `useState()` and `setState()`

`useState()` can accept a value or a callback into the [`initialState` argument](https://react.dev/reference/react/useState#usestate). These examples are the same: `const state = useState('foo')` OR `const state = useState(() => {return 'foo';})`.
The benefit of the 2nd example is that if you're doing something expensive the callback will only be executed on the first render, once the initialState has been set whatever is passed into this argument isn't 'needed' anymore by react. 

`setState()` accepts a `nextState` argument, which can be a value or updater function. [The updater is not always necessary.](https://react.dev/reference/react/useState#is-using-an-updater-always-preferred)

Reply:
Gotcha, I think in this case I don't want the `() =>` because the current `replayId` (renamed to `sessionId`) should be as up-to-date as possible on every initial render.